src/EFCore.PG.NTS/Storage/Internal/NpgsqlNetTopologySuiteTypeMappingSourcePlugin.cs

```diff
@@ -14,23 +15,92 @@ public class NpgsqlNetTopologySuiteTypeMappingSourcePlugin : IRelationalTypeMapp
+        };
+
         public NpgsqlNetTopologySuiteTypeMappingSourcePlugin([NotNull] INpgsqlNetTopologySuiteOptions options)
             => _options = Check.NotNull(options, nameof(options));
 
         public virtual RelationalTypeMapping FindMapping(in RelationalTypeMappingInfo mappingInfo)
         {
+            // TODO: Array
             var clrType = mappingInfo.ClrType;
             var storeTypeName = mappingInfo.StoreTypeName;
+            var isGeography = _options.IsGeographyDefault;
 
-            // TODO: Array
-            return clrType != null && typeof(IGeometry).IsAssignableFrom(clrType) ||
-                   storeTypeName != null && (
-                        storeTypeName.Equals("geometry", StringComparison.OrdinalIgnoreCase) ||
-                        storeTypeName.Equals("geography", StringComparison.OrdinalIgnoreCase))
+            if (clrType != null && !typeof(IGeometry).IsAssignableFrom(clrType))
+                return null;
+
+            if (storeTypeName != null)
+            {
+                if (!TryParseStoreTypeName(storeTypeName, out isGeography, out var parsedSubtype, out var _))
+                    return null;
+                if (clrType == null)
+                    clrType = parsedSubtype;
+            }
+
+            return clrType != null || storeTypeName != null
                 ? (RelationalTypeMapping)Activator.CreateInstance(
                     typeof(NpgsqlGeometryTypeMapping<>).MakeGenericType(clrType ?? typeof(IGeometry)),
-                    storeTypeName ?? (_options.IsGeographyDefault ? "geography" : "geometry"))
+                    storeTypeName ?? (isGeography ? "geography" : "geometry"),
+                    isGeography)
                 : null;
         }
+
+        /// <summary>
+        /// Given a PostGIS store type name (e.g. GEOMETRY, GEOGRAPHY(Point, 4326)), attempts to parse it and return its components.
+        /// </summary>
+        public static bool TryParseStoreTypeName(string storeTypeName, out bool isGeography, out Type clrType, out int srid)
+        {
+            isGeography = false;
+            clrType = null;
+            srid = -1;
+
+            var openParen = storeTypeName.IndexOf("(", StringComparison.Ordinal);
+
+            var baseType = (openParen > 0
+                ? storeTypeName.Substring(0, openParen)
+                : storeTypeName)
+                .Trim();
+
+            if (baseType.Equals("geometry", StringComparison.OrdinalIgnoreCase))
+                isGeography = false;
+            else if (baseType.Equals("geography", StringComparison.OrdinalIgnoreCase))
+                isGeography = true;
+            else
+                return false;
+
+            if (openParen == -1)
+                return true;
+
+            string subTypeString = null;
+            var closeParen = storeTypeName.IndexOf(")", openParen + 1, StringComparison.Ordinal);
+            if (closeParen > openParen)
```

Comment:
If there is a an open parenthesis then it goes before a closing one. Therefore, `closeParen` is `-1` or greater than `openParen`. So you can check that a closing parenthesis exists and return `false` if not.

Reply:
You're right, simplified the code.